packages/bvaughn-architecture-demo/src/hooks/useLocalStorage.ts

```diff
@@ -63,5 +80,14 @@ export default function useLocalStorage<T>(
     localStorageSetItem(key, string);
   }, [key, value]);
 
+  // Order of effects matters here. Don't trigger this until _after_ we've updated `localStorage`.
+  useLayoutEffect(() => {
+    // Only notify if this hook has had its setter run
+    if (updateCounter > 0) {
+      // Notify any other components subscribed to this value that it has changed
+      manuallyTriggerStorageEvent();
+    }
+  }, [updateCounter]);
```

Comment:
Can we just move this into the same hook as updates local storage?

https://github.com/replayio/devtools/blob/5a30eb174e7f8c4a39a6ee5031f817d4a44ebda4/packages/bvaughn-architecture-demo/src/hooks/useLocalStorage.ts#L76-L81

Basically
```js
localStorageSetItem(key, string);

manuallyTriggerStorageEvent();
```

Reply:
Maybe. I was trying to avoid any potential stringification render loop cases, as well as ensure that it only runs specifically once per attempt to set a value and _only_ in the component that had its setter called.

Reply:
I'm pretty sure what I suggested still meets those criteria, doesn't it?

Reply:
I dunno. Whatever. Either way is fine I guess.